src/jemalloc.c

```diff
@@ -1721,6 +1759,15 @@ malloc_init_hard_recursible(void) {
 	malloc_init_state = malloc_init_recursible;
 
 	ncpus = malloc_ncpus();
+	cpu_count_is_deterministic = malloc_cpu_count_is_deterministic();
+
+	if (!cpu_count_is_deterministic) {
```

Comment:
This condition is weak enough to instantly abort. If percpu arena wasn't enabled by the user, we shouldn't worry about CPU count. In fact, this caused one of the tests to abort unnecessarily:
https://app.travis-ci.com/github/jemalloc/jemalloc/jobs/552738103

Another thing is, if the user supplied `narenas` and enabled `percpu_arena`, they probably know what they're doing and should've set `nareanas` to a number large enough to accommodate the largest possible CPU ID. In this case, it might be worth it to just print a message saying `Warning: if narenas is less than the maximum CPU ID, you'll get undefined behavior`. This scenario is described here in the comment:
https://github.com/jemalloc/jemalloc/blob/dev/src/jemalloc.c#L1925-L1938

Only if the user asked for the percpu arena and didn't provide `narenas`, and this happened on a system that has a non-deterministic cpu count, we should either abort or change the setting (but also notify about the change in the setting).

Reply:
>If percpu arena wasn't enabled by the user, we shouldn't worry about CPU count

Indeed, nice catch, thanks, fixed!

>Another thing is, if the user supplied narenas and enabled percpu_arena, they probably know what they're doing and should've set nareanas to a number large enough to accommodate the largest possible CPU ID

Agree, added a message, though it is different, but I'm OK with using your message as-is, just let me know.